### ISSUE TITLE
table of contents ordered list

### ISSUE THREAD

**Issue by blaenk** [hoedown/hoedown#165]
I don't know if 4.0 will continue to have the table of contents generator (it'd be nice!), but one request I have is the ability to have it output ordered lists instead of unordered lists. Specifically, use `ol` instead of `ul` tags. This could probably be introduced as a flag, or something, since it's simply a one-character difference in the output, `TOC_ORDERED ? 'o' : 'u'` or something.

Currently I basically have to re-write my own toc renderer that is identical except for outputting `<ol>` instead of `<ul>`

**Comment by jmendeth** [comment]
I didn't plan on adding one in 4.0, mostly because there isn't a standard way to render a ToC in HTML. Each tool uses its own markup, some of what I've seen

~~~ html
<ul>
<li><a href="#toc1">1</a> Introduction</li>
<li><a href="#toc2">2</a> Blocks</li>
<li><a href="#toc2_1">2.1</a> Main components</li>
<li><a href="#toc2_2">2.2</a> Auxiliary components</li>
</ul>
~~~

~~~ html
<section>
<h1><span class="number">1</span>Introduction<a href="#toc1">#</a></h1>
<h1><span class="number">2</span>Blocks<a href="#toc2">#</a></h1>
<h2><span class="number">2.1</span>Main components<a href="#toc2_1">#</a></h2>
<h2><span class="number">2.2</span>Auxiliary components<a href="#toc2_2">#</a></h2>
</section>
~~~

~~~ html
<ol>
    <li><p class="entry"><a href="#toc1">Introduction</a></p>
    </li>
    <li><p class="entry"><a href="#toc2">Blocks</a></p>
        <ol>
            <li><p class="entry"><a href="#toc2_1">Main components</a></p>
            </li>
            <li><p class="entry"><a href="#toc2_2">Auxiliary components</a></p>
            </li>
        </ol>
    </li>
</ol>
~~~

It's not just one detail of the markup that varies, but the whole model. I'm not sure...

**Comment by uranusjr** [comment]
Maybe it would be better to create a wiki page for a list of “contrib” third-party components for this? The main repository can be shipped with an HTML renderer, maybe also an AST-tree renderer, but if someone builds custom renderers, or just some misc rendering callbacks, they can put a link to their repository (or maybe just a Gist) there, so those contrib packages can be more easily found by others. I have some extensions myself, like front-matter parsing, [Prism] integration, etc. They are probably not suitable for the core, but can be useful for specific needs.

[Prism]: http://prismjs.com

**Comment by jmendeth** [comment]
Yeah! I originally wanted to create a "Recipes" section in the wiki, but a dedicated repository can be a good idea too. 

(BTW, Prism integration? Don't we have that already?)

**Comment by uranusjr** [comment]
Oh, I was talking about direct syntax highlighting in the renderer, not post-render JavaScript highlighting, which is bad in performance. Basically you can run a headless JavaScript context (I use JavaScriptCore on OS X), and output the highlighted code directly.

**Comment by jmendeth** [comment]
Aaaah...
